Sources/SourceControl/GitRepository.swift

```diff
@@ -203,14 +204,31 @@ public class GitRepository: Repository, WorkingCheckout {
     /// The path of the repository on disk.
     public let path: AbsolutePath
 
+    /// The (serial) queue to execute git cli on.
+    private let queue = DispatchQueue(label: "com.apple.swiftpm.gitqueue")
+
     public init(path: AbsolutePath) {
         self.path = path
     }
 
     // MARK: Repository Interface
 
-    public var tags: [String] { return tagsCache.getValue(self) }
-    private var tagsCache = LazyCache(getTags)
+    /// Returns the tags present in repository.
+    public var tags: [String] {
```

Comment:
No, I don't think it has to change — clarifying it in the comment would be fine.  It's unfortunate that there is no shorthand syntax for sets, since it makes arrays be the collection that people end up using everywhere, even when conceptually the collection should really be unordered.